rust/lance/src/dataset.rs

```diff
@@ -5768,6 +5768,189 @@ mod tests {
         assert_eq!(results.num_rows(), 1);
     }
 
+    #[tokio::test]
+    async fn test_fts_unindexed_data_on_empty_index() {
+        let test_uri = TempStrDir::default();
+
+        // Empty dataset with fts index
+        let params = InvertedIndexParams::default();
+        let title_col = GenericStringArray::<i32>::from(Vec::<&str>::new());
+        let content_col = GenericStringArray::<i32>::from(Vec::<&str>::new());
+        let batch = RecordBatch::try_new(
+            arrow_schema::Schema::new(vec![
+                Field::new("title", title_col.data_type().to_owned(), false),
+                Field::new("content", title_col.data_type().to_owned(), false),
+            ])
+            .into(),
+            vec![
+                Arc::new(title_col) as ArrayRef,
+                Arc::new(content_col) as ArrayRef,
+            ],
+        )
+        .unwrap();
+        let schema = batch.schema();
+        let batches = RecordBatchIterator::new(vec![batch].into_iter().map(Ok), schema);
+        let mut dataset = Dataset::write(batches, &test_uri, None).await.unwrap();
```

Comment:
I'd recommend for most tests to just use the in-memory store:
```suggestion
        let mut dataset = Dataset::write(batches, "memory://", None).await.unwrap();
```